lib/Frenetic_NetKAT_Virtual_Compiler.ml

```diff
@@ -1,531 +1,13 @@
-    | _ -> false
-  in
-
-  let get_path_and_distance pv1 pv2 =
-    if is_loop pv1 pv2 then Some ([],0) else
-    match Tbl.find dist_tbl (pv1, pv2) with
-    | None ->
-      begin try
-        let path', dist = Dijkstra.shortest_path pgraph pv1 pv2 in
-        let path = unwrap_path path' in
-        Tbl.set dist_tbl ~key:(pv1, pv2) ~data:(path, dist);
-        Some (path, dist)
-      with Not_found ->
-        None
-      end
-    | pd -> pd
-  in
-
-  let path_oracle pv1 pv2 =
-    match get_path_and_distance pv1 pv2 with
-    | Some (p,d) -> p
-    | None -> assert false
-  in
-
-  let pv_of_v v =
-    match G.Prod.V.label v with
-    | InconsistentIn (_, pv) | InconsistentOut (_, pv)
-    | ConsistentIn (_, pv) | ConsistentOut (_, pv) -> pv
-  in
-
-  let cost v1 v2 =
-    match get_path_and_distance (pv_of_v v1) (pv_of_v v2) with
-    | Some (p,d) -> Some p
-    | None -> None
-  in
-
-  let pruned_graph = lazy (prune_product_graph prod_graph) in
-  let fabric_graph = lazy (fabric_graph_of_pruned (Lazy.force pruned_graph) prod_ing cost) in
-  let fabric = lazy (fabric_of_fabric_graph ?record_paths (Lazy.force fabric_graph) prod_ing path_oracle) in
-  let vg_file = "vg.dot" in
-  let pg_file = "pg.dot" in
-  let g_raw_file = "g_raw.dot" in
-  let g_pruned_file = "g_pruned.dot" in
-  let g_fabric_file = "g_fabric.dot" in
-  let vg_ch = open_out vg_file in
-  let pg_ch = open_out pg_file in
-  let g_raw_ch = open_out g_raw_file in
-  let g_pruned_ch = open_out g_pruned_file in
-  let g_fabric_ch = open_out g_fabric_file in
-  begin
-    if log then (
-      Printf.printf "[virtual] Statistics:\n";
-      Printf.printf "  |V(vgraph)|: %i\n" (G.Virt.nb_vertex vgraph);
-      Printf.printf "  |E(vgraph)|: %i\n" (G.Virt.nb_edges vgraph);
-      G.Virt.Dot.output_graph vg_ch vgraph;
-      close_out vg_ch;
-      Printf.printf "  |V(pgraph)|: %i\n" (G.Phys.nb_vertex pgraph);
-      Printf.printf "  |E(pgraph)|: %i\n" (G.Phys.nb_edges pgraph);
-      G.Phys.Dot.output_graph pg_ch pgraph;
-      close_out pg_ch;
-      Printf.printf "  |V(prod_graph)|: %i\n" (G.Prod.nb_vertex prod_graph);
-      Printf.printf "  |E(prod_graph)|: %i\n" (G.Prod.nb_edges prod_graph);
-      G.Prod.Dot.output_graph g_raw_ch prod_graph;
-      close_out g_raw_ch;
-      Printf.printf "  |V(pruned_graph)|: %i\n" (G.Prod.nb_vertex (Lazy.force pruned_graph));
-      Printf.printf "  |E(pruned_graph)|: %i\n" (G.Prod.nb_edges (Lazy.force pruned_graph));
-      G.Prod.Dot.output_graph g_pruned_ch (Lazy.force pruned_graph);
-      close_out g_pruned_ch;
-      Printf.printf "  |V(fabric_graph)|: %i\n" (G.Prod.nb_vertex (Lazy.force fabric_graph));
-      Printf.printf "  |E(fabric_graph)|: %i\n" (G.Prod.nb_edges (Lazy.force fabric_graph));
-      G.Prod.Dot.output_graph g_fabric_ch (Lazy.force fabric_graph);
-      close_out g_fabric_ch;
-      Printf.printf "\n");
-    Lazy.force fabric
-  end
-
-
+module Make(FG:FABRIC_GEN) =
+struct
 (*
```

Comment:
You can `include FG` here. That way you don't have to redefine `generate_fabric` ...